### PR TITLE
adding swap mirroring option to qscout_compile with tests

### PR DESCRIPTION
**Pull request by AndrewLitteken**
Discussed previously, swap mirroring is not the most common, and while it's good to have on by default, it may not be the expectation, and a user may have reasons to turn it off (explicit testing, or measurement outcomes).  This adds an option to cirq-superstaq to make it user configurable.

### CODE REVIEW

**Comment by AndrewLitteken** [comment]
Should the tests in the pull request also be adjusted as they are in the qiskit superstaq pull request?

**Comment by richrines1** [comment]
> Should the tests in the pull request also be adjusted as they are in the qiskit superstaq pull request?

yeah i believe so (imo the `qss` and `css` tests should be as similar as possible)

File: cirq_superstaq/service.py [hunk 1]
```diff
@@ -367,9 +371,15 @@ def qscout_compile(
         serialized_circuits = css.serialization.serialize_circuits(circuits)
         circuits_is_list = not isinstance(circuits, cirq.Circuit)
 
+        options_dict = {"swap_mirror": mirror_swaps}
```
Comment (richrines1) [inline cirq_superstaq/service.py:374]: nit
```suggestion
        options_dict = {"mirror_swaps": mirror_swaps}
```

File: cirq_superstaq/service_test.py [hunk 2]
```diff
@@ -318,6 +319,31 @@ def test_service_qscout_compile_single(mock_qscout_compile: mock.MagicMock) -> N
...
+        "jaqal_programs": [jaqal_program],
+    }
+
+    service = css.Service(api_key="key", remote_host="http://example.com")
+    out = service.qscout_compile(circuit, mirror_swaps=mirror_swaps)
+    assert out.circuit == circuit
+    assert out.jaqal_program == jaqal_program
+    mock_qscout_compile.assert_called_once()
+    assert json.loads(mock_qscout_compile.call_args[0][0]["options"]) == {
+        "swap_mirror": mirror_swaps
```
Comment (richrines1) [inline cirq_superstaq/service_test.py:343]:
```suggestion
        "mirror_swaps": mirror_swaps
```

**Review by richrines1 — COMMENTED** [review]
lg aside from a couple nits (and the updated `gss` requirements needed to make the check pass)

**Review by richrines1 — APPROVED** [review]
lgtm!

**Pull request merged**